### PR TITLE
Made descriptions editable on the search results view

### PR DESCRIPTION
**Pull request by amir-qayyum-khan**
Hi

In this PR i added inline editing feature to description of  learning resources

issue https://github.com/mitodl/lore/issues/389

@pdpinch @carsongee @noisecapella 

screen shot 2015-07-31 at 6 52 57 pm

screen shot 2015-07-31 at 6 53 37 pm

screen shot 2015-07-31 at 6 53 43 pm

screen shot 2015-07-31 at 6 53 55 pm

### CODE REVIEW

**Comment by noisecapella** [comment]
We should figure out some sort of UI mockup for how the edit button looks before we proceed further on this. It looks inconsistent with the rest of LORE

**Comment by amir-qayyum-khan** [comment]
hmmm, sure, @noisecapella  Do i need to create mockups or is there any designer on your team who will do that?

File: ui/static/ui/css/mit-lore.css [hunk 1]
```diff
@@ -451,3 +451,7 @@ ul.repos {
     padding: 0px;
     margin: 0px;
 }
+
```
Comment (noisecapella) [inline ui/static/ui/css/mit-lore.css:454]: If we are toggling this in javascript we might as well just set this value there on initialization.

File: ui/static/ui/js/listing.js [hunk 2]
```diff
@@ -76,7 +76,52 @@ define('listing',
         });
     }
 
+    function editDescLearningResource(lrId , repoSlug,  editButton) {
```
Comment (noisecapella) [inline ui/static/ui/js/listing.js]: These two functions need to be annotated with JSDoc

File: ui/static/ui/css/mit-lore.css (hunk 1 above)
Comment (amir-qayyum-khan) [inline ui/static/ui/css/mit-lore.css:454]: @noisecapella which  value , i am not able to understand, please explain

**Comment by amir-qayyum-khan** [comment]
Done @noisecapella @pdpinch 
- [x] Edit should be the  "pen" as seen here http://pureui.com/c/mit/lore/build004/ (click "manage taxonomies")
- [x] The edit form should be a 'bootstrap' form (see other forms for markup)
- [x] Try to make the width of the edit box  100%
- [x] In addition to 'save' the edit box needs a cancel button 
- [x] Clicking 'save' curently reloads the page. 
- [x] Only one edit box/form should be open at a given time.
- [x] Clicking "edit" on a lerning resources, should "cancel" other open edit forms.

screen shot 2015-08-03 at 3 37 14 pm

screen shot 2015-08-03 at 3 37 27 pm

screen shot 2015-08-03 at 3 37 39 pm

**Comment by carsongee** [comment]
It looks like the elastic search server failed during testing. I restarted the test to see if that will fix it.  You can as well by pressing the rebuild button on the build details:
image

**Comment by amir-qayyum-khan** [comment]
thanks @carsongee , i will it keep in mind.

**Comment by amir-qayyum-khan** [comment]
@carsongee it is fixed now. Thanks

**Comment by noisecapella** [comment]
I think this should get closed, after the #474 this implementation will be completely different

**Comment by pdpinch** [comment]
@Ferdi is it OK to close this an redo it after George gets back and wraps up #454?

I think it can wait.

**Comment by Ferdi** [comment]
I agree.

**Comment by pdpinch** [comment]
Closing this for now, until #454 is completed.

**Pull request closed without merging**